code/game/machinery/item_transporter.dm

```diff
@@ -0,0 +1,156 @@
+var/global/list/current_transporters = list()
+
+/obj/machinery/item_transporter
+	name = "transporter"
+	desc = "A machine that transports goods throughout the station."
+	icon = 'icons/obj/machines/item_transporter.dmi'
+	//state 0 is closed, state 1 is open
+	//state 01 is opening, state 10 is closing
+	icon_state = "transporter0"
+	//should be able to walk on it, but not dragged
+	anchored = TRUE
+	density = FALSE
+	//needs a cooldown, we aren't free
+	var/timed_cooldown = 0
+	//how much time between each transport
+	var/cooldown_time = 40 SECONDS
+	var/spawn_cooldown = 4 SECONDS
+	//required for deconstructing
+	var/decon_stage = 0
+	//so that multiple people can't use it
+	var/busy = FALSE
+
+/obj/machinery/item_transporter/examine(mob/user)
+	. = ..()
+	if(stat & NOPOWER)
+		to_chat(user, "<span class='info'>\The [src] is currently unpowered, please power it.</span>")
+	if(stat & BROKEN)
+		to_chat(user, "<span class='info'>\The [src] is currently broken, please repair it.</span>")
+	if(world.time < timed_cooldown)
+		var/time_remaining = round((timed_cooldown - world.time) / 10)
+		to_chat(user, "<span class='info'>\The [src] is currently on cooldown, please wait [time_remaining] seconds.</span>")
+
+/obj/machinery/item_transporter/New()
+	. = ..()
+	var/area/src_area = get_area(src)
+	var/rand_num = rand(100000, 999999)
+	name = "[src_area.name] [initial(name)] ([rand_num])"
+	current_transporters += src
+
+	component_parts = newlist(
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/micro_laser
+	)
+	RefreshParts()
+
+/obj/machinery/item_transporter/Destroy()
+	current_transporters -= src
+	. = ..()
+
+/obj/machinery/item_transporter/RefreshParts()
+	var/mb_rating = 0
+	var/l_rating = 0
+	cooldown_time = 40 SECONDS
+	spawn_cooldown = 5 SECONDS
+	for(var/obj/item/weapon/stock_parts/matter_bin/spMatterBin in component_parts)
+		mb_rating += spMatterBin.rating
+	for(var/obj/item/weapon/stock_parts/micro_laser/spLaser in component_parts)
+		l_rating += spLaser.rating
+	cooldown_time -= (mb_rating * 10) SECONDS
+	spawn_cooldown -= l_rating SECONDS
+
+/obj/machinery/item_transporter/attack_ghost(mob/user)
+	if(isAdminGhost(user))
+		var/choice_list = list(
+			"Reset Cooldown",
+			"Set Cooldown"
+		)
+		var/choice = input(user, "Choose which action to do") as null|anything in choice_list
+		switch(choice)
+			if("Reset Cooldown")
+				timed_cooldown = 0
+				return
+			if("Set Cooldown")
+				var/num_choice = input(user, "How many seconds should the cooldown be?") as num
```

Comment:
instead of all these  input windows, a UI would probably be desirable

Reply:
oh god, a ui.